### ISSUE TITLE
Software rendering problems with multiple domains.

### ISSUE THREAD

**Issue by aowen87** [aowen87/BAR#2106]
These issues were reported by Matthew Wheeler at AWE.

email:
======
Iâd like to report a couple of VisIt issues that one of our users came across recently.
 
There appear to be two problems:
Initially we were not sure if they were related â but we now suspect them to be different problems. (?)
(The user was seeing both problems at the same time)
 
I have now managed to reproduce them in much simpler form with smaller and public domain datasets.
(See notes further down). Hopefully these will help someone who understands that area better than me to pin it down.
 
 
1)       Visible white lines/breaks, ârandomlyâ between some domains when using Parallel Software-Rendering.
 
User was running Client-Server, Software Rendering with a large 2D AMR dataset and trying to display a Pseudocolor Plot with a slightly transparent Mesh Plot superimposed. âWhite linesâ appear between some domains as you zoom in and out but disappear if you switch to Hardware Rendering or force the problem to run on a single processor (-np 1). The resultant images can look very bad with a large complicated dataset. 
I have shown that the dataset does not have to be AMR â other 2D multi-domain datasets also seems to exhibit the same problem if you look carefully â but they donât always look as bad.
 
The problem seems similar to the reported VisIt Bug #1054 â âVisIt generates plots with gaps in between domainsâ
Also possibly similar to Bug #1350 â although this seems to have a different cause (and has been fixed?).

I think I have now managed to reproduce the problem with a much simpler dataset so hopefully this may help in debugging - see later.
 
 
2)       AMR levels appearing coarser than the levels defined in the dataset, i.e. not all the finer levels being shown. 
Sometimes switching levels off and on again seems to help â but not always. 
Again switching to Non-Scalable Rendering (=Hardware Rendering) mode produces the correct result.
Plotting the Mesh isnât necessary to reproduce this problem.
See test datasets and example pictures.
 
 
We have tried experimenting with a whole bunch of factors .e.g.:
Whether SR is on or off Before you start the Plot.
Sometimes the first time you produce the Plot it is different to subsequent renderings
Adding extra Operators â doesnât seem to affect the problem.
Re-Opening the dataset - ? - Not sure if this affects it
Occurs in different versions of VisIt â discovered in 2.6.1 but seems to be present 
  in all the other versions that we still have installed  ~ 2.4.* - 2.7.0b
- Difficult to see what the common factors are.
 
User has a short term workaround to use Hardware Rendering for both problems â but there must come a point when the datasets will become too large to display in this manner.
 
 
Test Datasets
--------------------
I have used two datasets to illustrate the problems â which you should be able to get at:
 
multi_curv2d.silo
 - illustrates problem 1, 
A 2D Curvilinear Mesh database â part of VisIt test suite (?) 
Consists of 5 domains/blocks in an arc.
 
amr2d_wmrgtree.silo
 - illustrates problems 1 and 2
A 2D AMR dataset downloaded from https://wci.llnl.gov/codes/silo/examples.html
(This is not actually as simple a dataset as it could be â Iâd be interested if you can find a more straightforward AMR dataset that we could both gain access to)
 
 
 
To Reproduce
---------------------
 
 Problem 1 â Domain breaks:
 
 
In Parallel Software Rendering mode, running over multiple processors (np > 1)
 
Open multi_curv2d.silo (or amr2d_wmrgtree.silo)
 
Do a Pseudocolor Plot of an appropriate variable (say âuâ or âDensityâ)
 
Add a Mesh Plot
Now turn the Transparency of the Mesh lines down below 100% e.g. 16%
 
 
Observe the âflickeringâ white lines between domains as you zoom in and out.
 
Note that if you switch to Hardware Rendering (which would be the default for small datasets!) you would get quite a pleasing result â the Mesh would appear as a subtle hint superimposed on top of the colored surface.
This must say something about where the problem is occurring â it would suggest that the data is being read-in and manipulated correctly â the problem would appear to be further down the rendering pipeline.
 
 
 
Problem 2 - Coarse AMR levels:
 
Just try to plot a Pseudocolor Plot of the AMR dataset while in Multi-Processor, Software Rendering mode.
Note the chunky representation which is resolved when you switch to Hardware Rendering.
 
 
 
I am sending you some images illustrating the problems.
They are in pairs with a good/OK (HW) and a BAD (SR) version but otherwise using the same setup
 
Let me know if I can provide more information or need to clarify anything.

-----------------------REDMINE MIGRATION-----------------------
This ticket was migrated from Redmine. The following information
could not be accurately captured in the new ticket:

Original author: Brad Whitlock
Original creation: 06/05/2013 06:02 pm
Original update: 06/26/2014 01:51 pm
Ticket number: 1487
